src/core/ai/llm/validation.ts

```diff
@@ -1,62 +1,175 @@
-import { validateModelFileSupport, validateFileSupport } from './registry.js';
+import { validateModelFileSupport, getAllowedMimeTypes } from './registry.js';
 import { logger } from '../../logger/index.js';
+import type { ImageData, FileData } from './messages/types.js';
 
-export interface FileValidationResult {
-    isSupported: boolean;
-    fileType?: string;
-    error?: string;
+export interface InputValidationResult {
+    isValid: boolean;
+    errors: string[];
+    fileValidation?: {
+        isSupported: boolean;
+        fileType?: string;
+        error?: string;
+    };
+    imageValidation?: {
+        isSupported: boolean;
+        error?: string;
+    };
 }
 
-export interface FileValidationConfig {
+export interface ValidationLLMConfig {
     provider: string;
     model?: string;
 }
 
 /**
- * Validates if a file is supported by the current LLM configuration.
- * Uses model-specific validation when available, falls back to provider-level.
- * @param config The LLM configuration containing provider and optional model
- * @param mimeType The MIME type of the file to validate
- * @returns Validation result with support status and error details
+ * Input interface for comprehensive validation
+ */
+export interface ValidationInput {
+    text?: string;
+    imageData?: ImageData | undefined;
+    fileData?: FileData | undefined;
+}
+
+/**
+ * Validates all inputs (text, image, file) against LLM capabilities and security requirements.
+ * This is the single entry point for all input validation.
+ * @param input The input data to validate (text, image, file)
+ * @param config The LLM configuration (provider and model)
+ * @returns Comprehensive validation result
  */
-export function validateFileForLLM(
-    config: FileValidationConfig,
-    mimeType: string
-): FileValidationResult {
+export function validateInputForLLM(
```

Comment:
`filterMessages` func has good coverage but we need some tests for this func as well as it's a critical point